File: apps/pulse-api/internal/domain/expenses/expenses.go
package expenses

type Expense struct {
	Id             uint    `json:"id"`
	ProductionName string  `json:"production_name"`
	ProductionId   uint    `json:"production_id"`
	Month          string  `json:"month"`
	Year           string  `json:"year"`
	TotalAmount    float64 `json:"total_deduction"`
} //@name Expense

type ExpenseDoc struct {
	Id        uint   `json:"id"`
	ExpenseId uint   `json:"-"`
	DocName   string `json:"name"`
	DocPath   string `json:"path"`
}

func (d Expense) Track() bool {
	return true
}

File: apps/pulse-web/src/server/types/expenses.ts
/*
 *Represent the expenses in expenses session
 */
export type Expenses = {
  id: number;
  production_name: string;
  year: string;
  month: string;
  production_id: number;
  total_deduction: number;
};

/*
 * Represent a new expense
 */

export type NewExpenseDTO = {
  files: File[];
  month: number;
  production_id: number;
  total_deduction: number;
};

export type Expense = NewExpenseDTO & {
  id: number;
  year: string;
  production_name: string;
};
